### PR TITLE
RHIROS-34 - apply RBAC permissions while rendering

### PR DESCRIPTION
**Pull request by kgaikwad**
For reference, adding below screenshots:
1. Systems List:
Screenshot from 2021-07-02 12-22-56

2. System Detail Page:
Screenshot from 2021-07-02 12-23-24

For ROS, 
- Two roles available 1. Resource Optimization administrator 2. Resource Optimization user
- Currently includes only wildcard permissions against resource. That means User won't be able to create custom roles on these permissions.
- By default, for both roles it adds read permission for inventory application i.e "inventory:*:read" so I have only checked ros side read permission wherever required.

Here is rbac config PR - https://github.com/RedHatInsights/rbac-config/pull/174

+ adding @karelhala & @terezanovotna into loop for UI review

### CODE REVIEW

**Comment by kgaikwad** [comment]
+ adding @johnsonm325 for review as I referred Drift side RBAC implementation

**Comment by terezanovotna** [comment]
Looks great. 
Slight changes to wording:

"You do not have access to Resource Optimization"

Add blue action button below "Go to landing page"
Link takes the user to the dashboard. 

More guidance on empty state usage can be found here: https://docs.google.com/document/d/1K6q0rQpOc9qOetHp1Zx7rHpnbKpPpDCfGIjhv6pja0U/edit#

**Comment by kgaikwad** [comment]
Thank you @terezanovotna for inputs.
With `NotAuthorized` component, code looks clean and much better. 
Removed unwanted code changes and made use of `NotAuthorized` component.


Screenshot from 2021-07-02 15-19-15

**Review by johnsonm325 — APPROVED** [review]
Everything looks good to me!

File: src/App.js [hunk 1]
```diff
@@ -26,22 +55,39 @@ class App extends Component {
...
             systemRecsReducer });
         insights.chrome.init();
-        // TODO change this to your appname
         insights.chrome.identifyApp('ros');
-
         this.appNav = insights.chrome.on('APP_NAVIGATION', event => this.props.history.push(`/${event.navId}`));
+
+        (async () => {
+            const rosPermissions = await insights.chrome.getUserPermissions('ros');
+            const permissionsList = rosPermissions.map(permissions => permissions.permission);
```
Comment (karelhala) [inline src/App.js:63]: Extra roundtrip. On line 65 you can do directly `permissionsList.some(({ permission }) => this.hasPermission(permission, ['ros:*:*', 'ros:*:read']))`

**Comment by kgaikwad** [comment]
Thank you @johnsonm325 & @karelhala for review.
Updated changes as per suggestions.

**Review by karelhala — APPROVED** [review]

**Review by apuntamb — APPROVED** [review]
Tested locally. Works as expected.

**Comment by apuntamb** [comment]
Merged!! Thanks @kgaikwad @terezanovotna  @karelhala @johnsonm325 !!

**Pull request merged**